prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import boto3
+import botocore
+from typing import List
+from prometheus_client.core import GaugeMetricFamily, CounterMetricFamily
+
+
+class GuardDutyMetricsCollector():
+    def __init__(self, regions: List[str]):
+        self.regions = regions
+        self.scrapeErrors = {region: 0 for region in regions}
+
+    def collect(self):
+        # Init metrics
+        currentFindingsMetric = GaugeMetricFamily(
+            "aws_guardduty_current_findings",
+            "The current number of unarchived findings",
+            labels=["region", "severity"])
+
+        scrapeErrorsMetric = CounterMetricFamily(
+            "aws_guardduty_scrape_errors_total",
+            "The total number of scrape errors",
+            labels=["region"])
+
+        # Customize the boto client config
+        botoConfig = botocore.client.Config(connect_timeout=2, read_timeout=10, retries={"max_attempts": 2})
+
+        for region in self.regions:
```

Comment:
If we do this using threads I'm pretty sure we could speed things up.

Reply:
I agree. Let me add it to our TODO list.